appengine/firebase-event-proxy/gae-firebase-event-proxy/src/main/java/com/example/GaeFirebaseEventProxy/FirebaseEventProxy.java

```diff
@@ -73,7 +70,7 @@ public void onDataChange(DataSnapshot snapshot) {
         if (snapshot.exists()) {
           try {
             // Convert value to JSON using Jackson
-            String json = new ObjectMapper().writeValueAsString(snapshot.getValue());
+            String json = new ObjectMapper().writeValueAsString(snapshot.getValue(false));
 
             // Replace the URL with the url of your own listener app.
             URL dest = new URL("http://gae-firebase-listener-python.appspot.com/log");
```

Comment:
I agree that what you suggest is better style. However this code will be used in an article as sample code and I think it is better to use code that directly calls Firebase API directly even though that hurts encapsulation. I think directly calling the API makes it easier for customers to understand the Firebase API because there are fewer things for them to be thinking about.

An early draft of the article is here:
https://docs.google.com/document/d/1H_W9C8cO_laKYElzbBBgIv9ECMBA3LRJqEebqfquWAI/edit#